### PR TITLE
Feat/validate json update

### PR DESCRIPTION
**Pull request by Vikka**
## What I am changing

- How the validation is done for `Geometry` to increase readability of specific type validation.

## How I did it

- Using `functools.singledispatchmethod`.

## How you can test it

- Run already existing tests.

## Important notes
This is for `Python >= 3.9`. https://github.com/python/cpython/issues/83860

### CODE REVIEW

File: geojson_pydantic/geometries.py [hunk 1]
```diff
@@ -6,6 +6,7 @@
 
 from pydantic import BaseModel, Field, ValidationError, validator
 from pydantic.error_wrappers import ErrorWrapper
+from singledispatchmethod import singledispatchmethod
```
Comment (vincentsarago) [inline geojson_pydantic/geometries.py:9]: this needs to be optional 

https://github.com/developmentseed/titiler/blob/master/src/titiler/core/titiler/core/factory.py#L64-L68

**Comment by vincentsarago** [comment]
> Error: The operation was canceled. for 3.9 and 3.8

is because the tests fails directly for 3.7 so GitHub actions kills all the other matrices

**Comment by Vikka** [comment]
This libraries fails on 3.7. I will investigate, but later this week.

**Comment by vincentsarago** [comment]
closing this. See https://github.com/developmentseed/geojson-pydantic/discussions/66#discussioncomment-3294233

**Pull request closed without merging**